src/video/wayland/SDL_waylandvideo.c

```diff
@@ -747,6 +747,17 @@ Wayland_free_display(uint32_t id)
         display = SDL_GetDisplay(i);
         data = (SDL_WaylandOutputData *) display->driverdata;
         if (data->registry_id == id) {
+            if (d->output_list != NULL) {
+                SDL_WaylandOutputData *node = d->output_list;
+                if (node == data) {
+                    d->output_list = node->next;
+                } else {
+                    while (node->next != data) {
```

Comment:
Not that I expect it to happen, but might be good to have an escape path:
```suggestion
                    while (node->next != data && node->next != NULL) {
```
Aside from that lgtm.

Reply:
Yeah that might be a useful safeguard.
I'd also have to add a check for `node->next = node->next->next;` tho.

Reply:
Yeah, probably going to have all the usual fun linked list boilerplate... wonder if that common linked list logic would be helpful at some point.